addons/mail/models/mail_blacklist.py

```diff
@@ -138,35 +122,33 @@ def _search_is_blacklisted(self, operator, value):
             else:
                 raise NotImplementedError()
 
-        [email_field] = self._primary_email
         if value:
             query = """
                 SELECT m.id
                     FROM mail_blacklist bl
                     JOIN %s m
-                    ON (LOWER(substring(m.%s, '([^ ,;<@]+@[^> ,;]+)')) = bl.email AND bl.active)
+                    ON m.email_normalized = bl.email AND bl.active
             """
         else:
             query = """
                 SELECT m.id
                     FROM %s m
                     LEFT JOIN mail_blacklist bl
-                    ON (LOWER(substring(m.%s, '([^ ,;<@]+@[^> ,;]+)')) = bl.email AND bl.active)
+                    ON m.email_normalized = bl.email AND bl.active
                     WHERE bl.id IS NULL
             """
-        self._cr.execute(query % (self._table, email_field))
+        self._cr.execute(query % self._table)
         res = self._cr.fetchall()
         if not res:
             return [(0, '=', 1)]
         return [('id', 'in', [r[0] for r in res])]
 
-    @api.depends(lambda self: self._primary_email)
+    @api.depends('email_normalized')
     def _compute_is_blacklisted(self):
         self._assert_primary_email()
-        [email_field] = self._primary_email
         # TODO : Should remove the sudo as compute_sudo defined on methods.
         # But if user doesn't have access to mail.blacklist, doen't work without sudo().
-        sanitized = [self.env['mail.blacklist']._sanitize_email(email) for email in self.mapped(email_field)]
-        blacklist = set(self.env['mail.blacklist'].sudo().search([('email', 'in', sanitized)]).mapped('email'))
+        blacklist = set(self.env['mail.blacklist'].sudo().search([
+            ('email', 'in', [email for email in self.mapped('email_normalized')])]).mapped('email'))
```

Comment:
`...search([('email',, 'in', self.mapped('email_normlized')])`should be enough. No need to create a new list [email for email in self ....]